packages/request-client.js/src/http-metamask-data-access.ts

```diff
@@ -0,0 +1,306 @@
+import { Block } from '@requestnetwork/data-access';
+import { requestHashSubmitterArtifact } from '@requestnetwork/smart-contracts';
+import { DataAccessTypes } from '@requestnetwork/types';
+import Utils from '@requestnetwork/utils';
+import axios, { AxiosRequestConfig } from 'axios';
+import { ethers } from 'ethers';
+
+// Maximum number of retries to attempt when http requests to the Node fail
+const HTTP_REQUEST_MAX_RETRY = 3;
+
+// Delay between retry in ms
+const HTTP_REQUEST_RETRY_DELAY = 100;
+
+/**
+ * Exposes a Data-Access module over HTTP
+ */
+export default class HttpMetamaskDataAccess implements DataAccessTypes.IDataAccess {
+  /**
+   * Cache block persisted directly (in case the node did not have the time to retrieve it)
+   * (public for easier testing)
+   */
+  public cache: {
+    [channelId: string]: {
+      [ipfsHash: string]: { block: DataAccessTypes.IBlock; storageMeta: any } | null;
+    };
+  } = {};
+
+  /**
+   * Configuration that will be sent to axios for each request.
+   * We can also create a AxiosInstance with axios.create() but it dramatically complicates testing.
```

Comment:
just curious, why?

Reply:
I think it is from @romaric-juniet (I copied/past it)

Reply:
I think this complexifies a lot usage of Axios... it might be related to the mock, not sure. anyway, we can cleanup later...